docs/next-docs/code-conventions.md

```diff
@@ -23,3 +23,48 @@ TypeScript types should be defined inside `src/lib/types.ts` file. Note that som
 ### Interfaces
 
 TypeScript types should be defined inside `src/lib/interfaces.ts` file. Note that some pre-existent interfaces could be defined in other files and will be temporarily kept there during the migration, to facilitate synchronization. These interfaces should be moved to `src/lib/interfaces.ts` later.
+
+### Component Props
+
+For all components, use the following pattern:
+
+```tsx
+type ComponentProps = {
+  // Prop types
+}
+
+// Destructure props at the declaration and use the name `props` when using the spread operator for the remaining props not specifically needed
+const Component = ({ title, label, ...props }: ComponentProps) => {
+  // Component code
+}
+
+/**
+ * Components using `forwardRef` from the Chakra UI package
+ *
+ * The first argument of the generic types is the props type signature.
+ *
+ * For the second argument of the generic types, you are declaring the primary element type that the component will render.
+ * This could be a `div`, `span`, `button`, etc. or a custom component (typeof Button) if said component is being used in the return.
+ */
+const Component = forwardRef<ComponentProps, "div">(
+  ({ title, label, ...props }, ref) => {
+    // Component code
+  }
+)
+```
```

Comment:
@wackerow @pettinarip 
I added clarifying text for the user to know exactly what is applied to the generic type arguments for `forwardRef`. what do you think about this addition?

Reply:
This is good for me. It is another use case where is useful to see our conventions applied to it.